tests/components/dyson/test_fan.py

```diff
@@ -50,33 +50,24 @@
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers import entity_registry
 
-from .common import ENTITY_NAME, NAME, SERIAL, async_update_device, get_basic_device
+from .common import (
+    ENTITY_NAME,
+    NAME,
+    SERIAL,
+    async_get_purecool_device,
+    async_get_purecoollink_device,
+    async_update_device,
+)
 
 ENTITY_ID = f"{PLATFORM_DOMAIN}.{ENTITY_NAME}"
 
 
 @callback
-def get_device(spec: Type[DysonPureCoolLink]) -> DysonPureCoolLink:
+def async_get_device(spec: Type[DysonPureCoolLink]) -> DysonPureCoolLink:
```

Comment:
This may return more than one type of dyson device instance, right? Do all have the same parent?

Reply:
This can only return `DysonPureCoolLink` or `DysonPureCool`. `DysonPureCool` is inherited from `DysonPureCoolLink`.

Reply:
Good!